### PR TITLE
carli/1937_cursor_info_for_spatial_profile_smoothing

### PR DESCRIPTION
**Pull request by crocka**
**Description**
A method called genSmoothedProfilerInfo is created. It extracts the y value of the smoothed profile and display the info below the spatial profiler.

**Checklist** 
- [x] changelog updated / ~~no changelog update needed~~ 
- [x] e2e test passing / ~~added corresponding fix~~ 
- [x] ~~protobuf updated to the latest dev commit~~ / no protobuf update needed

### CODE REVIEW

**Comment by kswang1029** [comment]
@crocka for the spatial profiler widget, we will also need to show the smoothed x value along with the smoothed y value too. Once it is added, a PR it is 👍 

Screen Shot 2022-09-16 at 08 32 53

**Review by kswang1029 — CHANGES_REQUESTED** [review]
@crocka thanks for adding the smoothed x value to the cursor info. We need to considered the smoothed x value as well (eg with binning, smoothed x is not necessary identical to the raw x).

https://user-images.githubusercontent.com/20819712/190954769-b0e35deb-d23e-4ac7-9337-dd4a22318aeb.mov

As for the smoothed wcs, it is a bit tricky I think due to the non-linear nature. I suggest we don't show smoothed wcs for now.

**Comment by kswang1029** [comment]
@crocka good for a PR 👍

File: src/components/SpatialProfiler/SpatialProfilerComponent.tsx [hunk 1]
```diff
@@ -413,7 +413,16 @@ export class SpatialProfilerComponent extends React.Component<WidgetProps> {
...
                     const xLabel = this.lineAxis ? `${this.lineAxis.label}: ${formattedExponential(nearest.point.x, 5)} ${this.lineAxis.unit ?? ""}, ` : `Image: ${nearest.point.x} px, `;
                     const valueLabel = `${nearest.point.y !== undefined ? formattedExponential(nearest.point.y, 5) : ""}`;
-                    profilerInfo.push("Cursor: (" + wcsLabel + xLabel + valueLabel + ")");
+
+                    const smoothedProfilerInfo = this.genSmoothedProfilerInfo(this.plotData?.smoothingValues);
+
+                    if (smoothedProfilerInfo && this.widgetStore.smoothingStore.isOverlayOn) {
+                        profilerInfo.push(`Cursor: (${wcsLabel}${xLabel}${valueLabel}, Smoothed: ${smoothedProfilerInfo})`);
+                    } else if (smoothedProfilerInfo) {
+                        profilerInfo.push(`Cursor: (${smoothedProfilerInfo})`);
```
Comment (YuHsuan-Hwang) [inline src/components/SpatialProfiler/SpatialProfilerComponent.tsx:456]: Do we also add `${wcsLabel}${xLabel}` here?

**Review by YuHsuan-Hwang — COMMENTED** [review]
overall looks good 👍

File: src/components/SpatialProfiler/SpatialProfilerComponent.tsx (hunk 1 above)
Comment (crocka) [reply src/components/SpatialProfiler/SpatialProfilerComponent.tsx:456]: Kuo-song said to leave wcsLabel and I have added xLabel to the smoothedProfilerInfo.

Comment (YuHsuan-Hwang) [reply src/components/SpatialProfiler/SpatialProfilerComponent.tsx:456]: Oh I see. Maybe we also remove `${wcsLabel}${imageLabel}` on L440 (at least `imageLabel`?)

Comment (crocka) [reply src/components/SpatialProfiler/SpatialProfilerComponent.tsx:456]: Yes! Sorry about that.

**Review by YuHsuan-Hwang — APPROVED** [review]

**Review by kswang1029 — APPROVED** [review]
manual test looks fine. no regression is observed from e2e tests. 👍

**Comment by YuHsuan-Hwang** [comment]
@crocka Please resolve conflicts and update changelog.

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 * Add a method to auto-scrolling the selected region into the region list view ([#1797](https://github.com/CARTAvis/carta-frontend/issues/1797)).
+* Added cursor info for smoothed spatial profile([#1938](https://github.com/CARTAvis/carta-frontend/pull/1938)).
```
Comment (YuHsuan-Hwang) [inline CHANGELOG.md]: I think this change is similar to the fourth item in Fixed: `Fixed issue where spetral profiler cursor info ...`. How about we combine them to, for example, `Show cursor info of smoothed profiles in the spatial and spectral profilers`

Comment (crocka) [reply CHANGELOG.md]: Thank you for pointing that out.

**Pull request merged**